idunn/geocoder/client.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import requests
+from fastapi import HTTPException
+from json.decoder import JSONDecodeError
+
+from idunn import settings
+from .models import GeocodeJson, QueryParams, ExtraParams
+
+
+logger = logging.getLogger(__name__)
+
+
+class GeocoderClient:
+    def __init__(self):
+        self.session = requests.Session()
+
+    def autocomplete(self, query: QueryParams, extra: ExtraParams):
+        url = settings["BRAGI_BASE_URL"] + "/autocomplete"
+
+        if extra.shape:
+            response = self.session.post(url, params=query.bragi_query_dict(), json=extra.dict())
+        else:
+            response = self.session.get(url, params=query.bragi_query_dict())
+
+        if response.status_code != requests.codes.ok:
+            try:
+                explain = response.json()["long"]
+            except (IndexError, JSONDecodeError):
+                explain = response.text
+
+            logger.error(
+                'Request to Bragi returned with unexpected status %d: "%s"',
+                response.status_code,
+                explain,
+            )
+            raise HTTPException(503, "Unexpected geocoder error")
+
+        return GeocodeJson.parse_obj(response.json())
```

Comment:
Shouldn't we put this into a try block?

Reply:
Just pushed https://github.com/QwantResearch/idunn/pull/105/commits/50d537e34a7c16afc2cfc8177c7440661f80a55e